subprojects/gradle/process/src/main/kotlin/com/avito/utils/ProcessRunner.kt

```diff
@@ -1,129 +1,147 @@
 package com.avito.utils
 
 import com.avito.android.Result
-import com.avito.logger.Logger
-import com.avito.logger.LoggerFactory
 import org.apache.tools.ant.types.Commandline
+import org.gradle.internal.impldep.com.google.common.annotations.VisibleForTesting
 import java.io.BufferedReader
 import java.io.File
 import java.io.InputStream
 import java.io.InputStreamReader
+import java.util.concurrent.Callable
+import java.util.concurrent.ExecutorService
+import java.util.concurrent.SynchronousQueue
+import java.util.concurrent.ThreadPoolExecutor
+import java.util.concurrent.TimeUnit
 
 interface ProcessRunner {
 
     /**
-     * @param command например ls -la
+     * @param command i.e. ls -la
      *
      * @return output команды, если exit code = 0
      */
+    fun run(command: String, timeout: Long, unit: TimeUnit): Result<String>
+
     fun run(command: String): Result<String>
 
+    fun spawn(command: String, outputTo: File?): Process
+
     class Real(
         private val workingDirectory: File?,
-        private val loggerFactory: LoggerFactory
+        private val cliOutputReaderThreadPool: ExecutorService =
```

Comment:
Pool for threads that reads cli output:
before -  2 new thread for each cli operation without caching
now - max 300 threads, keep alive time=30s. Zero impact after 30 sec idle